frontend/src/hocs/withHostCommunication/withHostCommunication.tsx

```diff
@@ -71,7 +103,22 @@ function withHostCommunication(
     const [deployedAppMetadata, setDeployedAppMetadata] = useState({})
     const [toolbarItems, setToolbarItems] = useState<IToolbarItem[]>([])
 
+    const [allowedOriginsResp, setAllowedOriginsResp] =
+      useState<IAllowedMessageOriginsResponse | null>(null)
+    const [deferredAuthToken, setDeferredAuthToken] = useState<
+      Resolver<string | undefined>
+    >(() => new Resolver())
+
     useEffect(() => {
+      if (!allowedOriginsResp) {
+        return () => {}
+      }
+
+      const { allowedOrigins, useExternalAuthToken } = allowedOriginsResp
+      if (!useExternalAuthToken) {
+        deferredAuthToken.resolve(undefined)
+      }
+
```

Comment:
I'm writing out my thoughts. This is essentially saying, when the allowedOrigins response changes (ie we get that response), we check whether the origins response informs us to use the external auth token. If it doesn't require one, we just resolve auth token to be `undefined` explicitly setting the token is not used.